strax/plugin.py

```diff
@@ -345,13 +345,21 @@ def iter(self, iters, executor=None):
                         self.input_buffer[d].split(
                             t=this_chunk_end,
                             allow_early_split=True)
-
                 # If any of the inputs were trimmed due to early splits,
                 # trim the others too.
                 # In very hairy cases this can take multiple passes.
                 # TODO: can we optimize this, or code it more elegantly?
                 max_passes_left = 10
                 while max_passes_left > 0:
+                    if int(self.save_when) == strax.SaveWhen.NEVER:
+                        # So this is a bit funny, this is the only time we may
+                        # be actually running a plugin that is not saved with
+                        # the time range. This means that we might be asking
+                        # for a time range that is inconsistent with the
+                        # chunking of the inputs. See:
+                        # https://github.com/AxFoundation/strax/issues/247
+                        break
```

Comment:
I think this triggers even during regular processing of a run, without a time range selection. Are you sure it is harmless? Maybe I'm not understanding why this break is needed; did we get the "ten passes" error message when plotting waveforms?

Reply:
Thanks, this was actually the first place I started looking. This was indeed not necessary and also felt slightly out of place. Fun fact, the input times were inconsistent (not the matter here). 

I removed it now.